File: C++/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int depth = 0;
        int maxDepth = 0;
        for (char c: s) {
            if (c == '(') depth++;
            if (c == ')') depth--;
            if (maxDepth < depth) maxDepth = depth;
        }
        return maxDepth;
    }
};

File: Go/maximum-nesting-depth-of-the-parentheses.go
func maxDepth(s string) int {
    result := 0
    depth := 0
    for _, c := range s {
        if c == '(' {
            depth++
        } else if c == ')' && depth > 0 {
            depth--
        }
        if depth > result {
            result = depth
        }
    }
    return result
}